### ISSUE TITLE
how to load binary files over http?

### ISSUE THREAD

**Issue by amitu** [gopherjs/gopherjs#196]
I am trying this:

```go
func LoadFile(filename string) ([]byte, error) {
	ch := make(chan error, 1)

	req := js.Global.Get("XMLHttpRequest").New()
	req.Call("overrideMimeType", "text/plain; charset=x-user-defined")

	req.Call("open", "GET", filename, true)
	req.Call(
		"addEventListener", "load", func(*js.Object) {
			go func() { ch <- nil }()
		}, false,
	)
	req.Call(
		"addEventListener", "error", func(o *js.Object) {
			go func() { ch <- &js.Error{Object: o} }()
		}, false,
	)
	req.Call("send", nil)

	err := <-ch
	if err != nil {
		return nil, err
	}

	resp := req.Get("responseText").String()
	fmt.Println("resp", len(resp))
	return []byte(resp), nil
}
```

I feel wrong about calling .String() on responseText there, as I am expecting binary data. 

The problem is its showing wrong size if the file is binary. On text file it works fine. On binary files it increasing the reported file size. Actual file size: 753938, fmt reports: 1509758.

**Comment by cryptix** [comment]
IIRC @neelance recently implemented the net/http client by wrapping XHR, so you shouldn't need to do this by yourself and just use `http.Get()` and read from the returned `response.Body`.

**Comment by cryptix** [comment]
Support landed on fdbe407578d08ab76f9150277f68d01e45ee5f16 If you want to see how to do it, but you wont have to in this case.

**Comment by amitu** [comment]
Thanks a lot cryptix. Works for me :-)
